asmcomp/asmlink.ml

```diff
@@ -213,7 +213,7 @@ let scan_file obj_name (tolink, objfiles) = match read_file obj_name with
           infos.lib_units tolink
       and objfiles =
         if infos.lib_units = []
-        && not (Sys.file_exists (object_file_name obj_name)) then
+        && not (Sys.file_exists file_name) then
```

Comment:
The value `file_name` returned by `read_file` always ends in `".cmx"` or `".cmxa"`, but that returned by `object_file_name` ends in `ext_obj` or `ext_lib`, so the two calls do not seem to be exactly equivalent. Am I missing something?

Reply:
No you are right, that checks the wrong file, same in the new `object_file_name_of_file` function. I reworked the patch set (only two commits now).